Fixes a problem when the keyboard is shown under iOS 7.

diegoreymendez opened the pull request:
Fixes [a problem when the keyboard is shown under iOS 7 after inserting images](https://github.com/wordpress-mobile/WordPress-iOS-Editor/issues/301).

WPiOS branch `issue/301-cant-scroll` integrates this fix and makes it easy to test it.

/cc @SergioEstevao and @bummytime

SergioEstevao commented on Classes/WPEditorView.m:
```diff
@@ -368,11 +368,63 @@ - (void)stopObservingKeyboardNotifications
 
 - (void)keyboardDidShow:(NSNotification *)notification
 {
+    BOOL isiOS7OrEarlier = ([[[UIDevice currentDevice] systemVersion] floatValue] < 8.0);
```
Maybe add an extension on UIDevice there is already one for detecting iOS8

bummytime replied:
What are your thoughts on moving this to ```WPDeviceIdentification``` as a stand-alone method?

diegoreymendez replied:
Good call.  I ended up adding a method to `WPDeviceIdentification` since that class has the intention of providing these methods.

bummytime commented on Classes/WPEditorView.m:
```diff
@@ -368,11 +368,63 @@ - (void)stopObservingKeyboardNotifications
+    if (isiOS7OrEarlier) {
+        // PROBLEM: under iOS 7, it seems that setting the proper insets in keyboardWillShow: is not
+        // enough.  We were having trouble when adding images, where the keyboard would show but the
+        // insets would be reset to {0, 0, 0, 0} between keyboardWillShow: and keyboardDidShow:
```
:weary: nice job with finding this.  I swear we need a blog post that enumerates all of the workarounds we have done!

bummytime commented:
@diegoreymendez I had one comment above...do what you want with that. The fix works well. :shipit:

diegoreymendez commented:
Changed the code following the comments above, and added some details to a comment to make it easier to reproduce the initial issue.

Ready for another :eyes:

bummytime commented:
:shipit:
